File: src/hello_world_ros/src/hello_world.cpp
// Copyright (c) 2017, David Ball
// License: BSD 3-clause. (See LICENSE file.)

#include "ros/ros.h"

int main(int argc, char **argv)
{
  ros::init(argc, argv, "hello_world");
  ros::NodeHandle n;

  ROS_DEBUG("Hello world! (debug)"); // won't appear due to default log level
  ROS_INFO("Hello world! (info)");
  ROS_WARN("Hello world! (warn)");
  ROS_ERROR("Hello world! (error)");

  ROS_DEBUG_STREAM("Hello world! " << "(debug)"); // won't appear due to default log level
  ROS_INFO_STREAM("Hello world! " << "(info)");
  ROS_WARN_STREAM("Hello world! " << "(warn)");
  ROS_ERROR_STREAM("Hello world! " << "(error)");

  return EXIT_SUCCESS;
}

File: src/hello_world_ros/src/hello_world.py
#!/usr/bin/env python
# Copyright (c) 2017, David Ball
# License: BSD 3-clause. (See LICENSE file.)
import rospy

def hello_world():
  rospy.init_node('hello_world', anonymous=True)
  rospy.sleep(1.0)
  rospy.logdebug("Hello world! (debug)") # Won't be displayed
  rospy.logwarn("Hello world! (warn)")
  rospy.loginfo("Hello world! (info)")
  rospy.logerr("Hello world! (error)")
  rospy.logfatal("Hello world! (fatal)")

if __name__ == '__main__':
  hello_world()
